ledger/src/blockstore.rs

```diff
@@ -8017,6 +8012,7 @@ pub mod tests {
 
         if simulate_ledger_cleanup_service {
             *blockstore.lowest_cleanup_slot.write().unwrap() = lowest_cleanup_slot;
+            blockstore.purge_slots(0, lowest_cleanup_slot, PurgeType::CompactionFilter);
```

Comment:
Did unit test fail without this ?

Reply:
Yes, because `Blockstore::first_available_block()` depends on the rooted-slot iterator, and this "simulation" wasn't adjusting the root list, just writing a new lowest_cleanup_slot. This change seemed defensible to me, since this is exactly what the LedgerCleanupService does, plus some extra data reporting.